src/sbd-pacemaker.c

```diff
@@ -83,6 +83,98 @@ pe_free_working_set(pe_working_set_t *data_set)
 
 #endif
 
+static void
+clean_up(int rc);
+
+#ifdef XML_PING_ATTR_PACEMAKERDSTATE
+static char *admin_uuid = NULL;
+static crm_ipc_t *pacemakerd_channel = NULL;
+time_t last_ok = (time_t) 0;
+
+static int
+pacemakerd_msg_callback(const char *buffer, ssize_t length, gpointer userdata);
+static void
+pacemakerd_ipc_connection_destroy(gpointer user_data);
+
+static struct ipc_client_callbacks pacemakerd_callbacks = {
+    .dispatch = pacemakerd_msg_callback,
+    .destroy = pacemakerd_ipc_connection_destroy
+};
+
+static void
+ping_pacemakerd(void)
+{
+    if (pacemakerd_channel) {
+        xmlNode *cmd = create_request(CRM_OP_PING, NULL, NULL,
+                                      CRM_SYSTEM_MCP, crm_system_name,
+                                      admin_uuid);
+
+        crm_ipc_send(pacemakerd_channel, cmd, 0, 0, NULL);
+        free_xml(cmd);
```

Comment:
This is the meat of the new API right here :) Then the scaffolding is 10 times as much. I would think we just need a pacemakerd client object with connect, disconnect, and ping methods, and new/free functions for that. Not sure what the naming should look like.